File: zapateria-mobile/src/services/categoria.service.ts
import api from '../lib/api';
import { Categoria } from '../types';

export interface CreateCategoriaDto {
  nombre: string;
  activo?: boolean;
}

export interface UpdateCategoriaDto {
  nombre?: string;
  activo?: boolean;
}

export const categoriaService = {
  getAll: async (): Promise<Categoria[]> => {
    const response = await api.get('/categorias');
    return response.data;
  },

  getOne: async (id: string): Promise<Categoria> => {
    const response = await api.get(`/categorias/${id}`);
    return response.data;
  },

  create: async (data: CreateCategoriaDto): Promise<Categoria> => {
    const response = await api.post('/categorias', data);
    return response.data;
  },

  update: async (id: string, data: UpdateCategoriaDto): Promise<Categoria> => {
    const response = await api.put(`/categorias/${id}`, data);
    return response.data;
  },

  delete: async (id: string): Promise<void> => {
    await api.delete(`/categorias/${id}`);
  },
};

File: zapateria_flutter/lib/services/categoria_service.dart
import 'package:dio/dio.dart';
import '../models/models.dart';
import 'api_client.dart';

class CategoriaService {
  final Dio _dio = apiClient.dio;

  Future<List<CategoriaModel>> getAll() async {
    final response = await _dio.get('/categorias');
    return (response.data as List).map((j) => CategoriaModel.fromJson(j)).toList();
  }

  Future<CategoriaModel> getOne(String id) async {
    final response = await _dio.get('/categorias/$id');
    return CategoriaModel.fromJson(response.data);
  }

  Future<CategoriaModel> create({required String nombre}) async {
    final response = await _dio.post('/categorias', data: {'nombre': nombre});
    return CategoriaModel.fromJson(response.data);
  }

  Future<CategoriaModel> update(String id, {String? nombre, bool? activo}) async {
    final data = <String, dynamic>{};
    if (nombre != null) data['nombre'] = nombre;
    if (activo != null) data['activo'] = activo;
    final response = await _dio.put('/categorias/$id', data: data);
    return CategoriaModel.fromJson(response.data);
  }

  Future<void> delete(String id) async {
    await _dio.delete('/categorias/$id');
  }
}

final categoriaService = CategoriaService();
